### PR TITLE
(OraklNode) Add github action for testing

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

- linter and vet checks
- minor updates to fix shadowing error and lint error found while writing github action

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: node/taskfiles/taskfile.local.yml [hunk 1]
```diff
@@ -9,3 +9,7 @@ tasks:
     dotenv: [".env"]
     cmds:
       - go test ./pkg/admin/tests -v
+  test:
```
Comment (martinkersner) [inline node/taskfiles/taskfile.local.yml:12]: You could use dependencies instead. It is more clear.

```
test:
  deps:
    - test-db
    - test-admin
```

File: node/pkg/raft/raft.go [hunk 2]
```diff
@@ -54,7 +57,10 @@ func (r *Raft) Run(ctx context.Context, node Node) {
 		case <-r.ElectionTimer.C:
 			r.startElection()
 		case <-jobTicker:
-			node.Job()
+			err := node.Job()
+			if err != nil {
+				log.Println("failed to execute job:", err)
```
Comment (martinkersner) [inline node/pkg/raft/raft.go:62]: At some point, you should start using log levels.

**Review by martinkersner — APPROVED** [review]
LGTM!

File: node/taskfiles/taskfile.local.yml (hunk 1 above)
Comment (nick-bisonai) [reply node/taskfiles/taskfile.local.yml:12]: deps are to define tasks that depend on each other, but in this case, it's not depending on each other, but I wanted to declare a task that runs all test tasks.
https://taskfile.dev/usage/#task-dependencies

in official document, it is recommending to use `cmds` to run other tasks
https://taskfile.dev/usage/#calling-another-task

**Pull request merged**